chainerx_cc/chainerx/context.cc

```diff
@@ -265,6 +255,19 @@ std::vector<BackpropId> Context::GetInnerBackpropIds(const BackpropId& backprop_
     return inner_backprop_ids;
 }
 
+Backend& Context::RegisterBackend(const std::string& backend_name, std::unique_ptr<Backend, context_detail::BackendDeleter> backend) {
+    Backend& backend_ref = *backend;
+    {
+        std::lock_guard<std::mutex> lock{mutex_};
+        auto pair = backends_.emplace(backend_name, std::move(backend));
+        if (!pair.second) {
+            ContextError{"Backend is already registered: ", backend_name};
```

Comment:
Previously we didn't throw in this case. We can't change this behavior (consider the case a backend is used in multiple threads).

Reply:
Fixed.